src/python/gem5/resources/downloader.py

```diff
@@ -79,38 +80,20 @@ def _download(url: str, download_to: str, max_attempts: int = 6) -> None:
 
     # TODO: This whole setup will only work for single files we can get via
     # wget. We also need to support git clones going forward.
-
     attempt = 0
     while True:
         # The loop will be broken on a successful download, via a `return`, or
         # if an exception is raised. An exception will be raised if the maximum
         # number of download attempts has been reached or if a HTTP status code
         # other than 408, 429, or 5xx is received.
         try:
-            # check to see if user requests a proxy connection
             use_proxy = os.getenv("GEM5_USE_PROXY")
             if use_proxy:
-                # If the "use_proxy" variable is specified we setup a socks5
-                # connection.
-
-                import socket
-                import ssl
-
-                import socks
-
-                IP_ADDR, host_port = use_proxy.split(":")
-                PORT = int(host_port)
-                socks.set_default_proxy(socks.SOCKS5, IP_ADDR, PORT)
-                socket.socket = socks.socksocket
-
-                # base SSL context for https connection
-                ctx = ssl.create_default_context()
-                ctx.check_hostname = False
-                ctx.verify_mode = ssl.CERT_NONE
-
                 # get the file as a bytes blob
                 request = urllib.request.Request(url)
-                with urllib.request.urlopen(request, context=ctx) as fr:
+                with urllib.request.urlopen(
+                    request, context=get_proxy_context()
+                ) as fr:
                     with tqdm.wrapattr(
                         open(download_to, "wb"),
                         "write",
```

Comment:
I'm curious as to why we've got two different ways to download depending on whether a proxy was specified or not. The only difference appears to be that if proxy is specified `urlopen` is used while if it's not `urlretrieve` is used. If we used the `urlopen` approach for both, would that not work?

**This is not a blocker for this PR. Nor am I seriously asking for you to investigate. This isn't your code. I just think it's an odd design and I don't know the reason**.

Reply:
This is my code, IIRC. I don't remember why. Hopefully I wrote something down in the commit message 😆